core/runtime/src/main/java/io/quarkus/runtime/configuration/MemorySize.java

```diff
@@ -25,4 +26,19 @@ public long asLongValue() {
     public BigInteger asBigInteger() {
         return value;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object)
+            return true;
+        if (object == null || getClass() != object.getClass())
+            return false;
+        MemorySize that = (MemorySize) object;
+        return Objects.equals(value, that.value);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(value);
```

Comment:
Store the hashCode as an attribute in the constructor for improved performance

Reply:
I don't expect it to be hot path so I would say we can avoid it.